mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/kafka/Consumer.java

```diff
@@ -1,16 +1,103 @@
 package de.brandwatch.minianalytics.mentiongenerator.kafka;
 
+import de.brandwatch.minianalytics.mentiongenerator.model.Mention;
+import de.brandwatch.minianalytics.mentiongenerator.model.Resource;
+import de.brandwatch.minianalytics.mentiongenerator.postgres.model.Query;
+import de.brandwatch.minianalytics.mentiongenerator.postgres.repository.QueryRepository;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.TextField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.RAMDirectory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.kafka.annotation.KafkaListener;
 
+import java.io.IOException;
+import java.util.List;
+
 public class Consumer {
 
     private static final Logger logger = LoggerFactory.getLogger(Consumer.class);
 
-    @KafkaListener(topics = "twitter")
-    public void receive(String message){
-        logger.info("received message='{}'", message);
+    private final QueryRepository queryRepository;
+
+    private final Producer producer;
+
+    private final Directory memoryIndex = new RAMDirectory();
+    private final StandardAnalyzer analyzer = new StandardAnalyzer();
+
+    private final IndexWriterConfig indexWriterConfig = new IndexWriterConfig(analyzer);
+    private final IndexWriter writer = new IndexWriter(memoryIndex, indexWriterConfig);
+
+    private IndexSearcher searcher;
+    private IndexReader reader;
+
+    private QueryParser queryParser = new QueryParser("text", analyzer);
+
+    @Autowired
+    public Consumer(QueryRepository queryRepository, Producer producer) throws IOException {
+        this.queryRepository = queryRepository;
+        this.producer = producer;
+    }
+
+    @KafkaListener(topics = "resources")
+    public void receive(Resource resource) throws ParseException, IOException {
+        logger.info("received message='{}'", resource.toString());
+
+        //Create Lucene Document out of Resource
+        Document document = new Document();
+
+        document.add(new TextField("author", resource.getAuthor(), Field.Store.YES));
```

Comment:
What exactly does `Field.Store.YES` do?

Reply:
> Indexed - The field is analyzed and indexed, and can be searched.
> Stored - The field's full text is stored and will be returned with search results.

I don't actually think I need it to be stored. I search for it and if I hit, then I will create a `Mention` out of the `Resource` Object and not out of the `Document`

Reply:
That's what I was getting at ;)

Reply:
I already changed it